src/policy/policy.cpp

```diff
@@ -23,22 +23,42 @@
 #include <cstddef>
 #include <vector>
 
+/**
+ * Calculate the dust threshold for an output based on the given fee rate.
+ *
+ * "Dust" is defined in terms of dustRelayFee, which has units satoshis-per-kilobyte.
+ * If you'd pay more in fees than the value of the output to spend something, then we consider it dust.
+ *
+ * Definitions for different output types:
+ * - A typical spendable non-segwit P2PKH txout is 34 bytes big, and will
+ *   need a CTxIn of at least 148 bytes to spend:
+ *   so dust is a spendable txout less than 182*dustRelayFee/1000 (in satoshis).
+ *   546 satoshis at the default rate of 3000 sat/kvB.
+ * - A typical spendable non-segwit P2SH txout is 32 bytes big, and will
+ *   need a CTxIn of at least 148 bytes to spend:
+ *   so dust is a spendable txout less than 180*dustRelayFee/1000 (in satoshis).
+ *   540 satoshis at the default rate of 3000 sat/kvB.
+ * - A typical spendable segwit P2WPKH txout is 31 bytes big, and will
+ *   need a CTxIn of at least 67 bytes to spend:
+ *   so dust is a spendable txout less than
+ *   98*dustRelayFee/1000 (in satoshis).
+ *   294 satoshis at the default rate of 3000 sat/kvB.
+ * - A typical spendable segwit P2TR txout is 43 bytes big, and will
+ *   need a CTxIn of at least 57（used 67, see reason below） bytes to spend:
+ *   so dust is a spendable txout less than
+ *   110*dustRelayFee/1000 (in satoshis).
+ *   330 satoshis at the default rate of 3000 sat/kvB.
```

Comment:
This specifically does not match the code.

Reply:
> This specifically does not match the code.

Here we want to explain the dust calculation logic of existing common scripts

The code below simplifies the logic of dust level and says in the comment that the dust limit will not be lowered for segwit v1

I think we need to explain the situation clearly, although we have some considerations in engineering to choose a general solution

Reply:
> This specifically does not match the code.
Here we want to explain the dust calculation logic of existing common scripts

The code below simplifies the logic of dust level and says in the comment that the dust limit will not be lowered for segwitv1

I think we need to explain the situation clearly, although we have some considerations in engineering to choose a general solution

Reply:
This explanation will make the entire dust calculation logic more complete and I think it should be considered carefully.

Reply:
I somewhat agree that if the code does something slightly different from what one could expect, it may be worth adding a comment - it would depend on the situation. Verbose discussions of engineering tradeoffs usually belong in discussion forums, docs, and bips, not function descriptions.

What we have currently seems fine to me: "Note this computation is for spending a Segwit v0... For Segwit v1... this computation was kept. See discussion... for more details". It is in a suitable location, i.e. where you might expect to find this code and wonder why it's not there.

If I were reading the comment added here as a caller, I'd expect that the calculation given here is exactly what the implementation does. Is the p2tr dust threshold is actually 330sat?

Reply:
OK, I agree with you. It may be more appropriate to put it in discuss. 

I will close this issue. Thank you for your review.